src/csl/quote.rs

```diff
@@ -0,0 +1,171 @@
+use citationberg::{taxonomy::OtherTerm, TermForm};
+
+use super::{taxonomy::EntryLike, Context};
+
+/// A smart quote substitutor with zero lookahead.
+/// TODO: Move to its own crate.
```

Comment:
I think this is fine as is. You can open a separate issue to discuss whether it'd make sense to use some shared impl between Typst and hayagriva, however.

```suggestion
```